### DISCUSSION TITLE
UE in a different machine

### DISCUSSION THREAD

**Discussion by dayal007** [my5G/my5G-non3GPP-access#10] [Ideas]
Hello,

What configuration changes, interfaces updates and firewall rules need to be done if we want to run the UE in a separate machine? Because that will be the most realistic scenario then running UE and core in the same machine.

Example

|UE - 192.168.1.50| -> |my5g-core - 192.168.1.55|

As of now, this expects everything need to be run in the same machine.

Is my understanding right?

On my5gcore machine:

  sudo ip link add name ipsec0 type vti local 192.168.1.55 remote 0.0.0.0  key 5
  sudo ip addr add 10.0.0.1/24 dev ipsec0
  sudo ip link set ipsec0 up

On UE Machine:

sudo ip link add ipsec0 type vti local 192.168.1.50 remote 192.168.1.55 key 5
sudo ip link set ipsec0 up

And then "trigger_initial_registration.sh"

ue_addr=${ue_addr:-"192.168.1.50"}
ue_port=${ue_port:-"10000"}
scheme=${scheme:-"https"}
auth_method=${auth_method:-"5G_AKA"}
n3iwf_address=${n3iwf_address:-"192.168.1.55"}
supi_or_suci=${supi_or_suci:-"2089300007487"}
k=${k:-"5122250214c33e723a5dd523fc145fc0"}
opc_type=${opc_type:-"OP"}
opc=${opc:-"c9e8763286b5b9ffbdf56e1297d0887b"}
ike_bind_addr=${ike_bind_addr:-"192.168.1.50"}

Thanks,
Udhay

**Comment by mariotlemes** [comment]
Hello,

I do not complete understand your question, but to usage the non-3GPP access all you need is found in [usage section](https://github.com/my5G/my5G-non3GPP-access/wiki/Usage).

If in your scenario the Y1 and Y2 interfaces are correctly established, you can use the following options for run the non-3GPP access:

Options:
  --ue_addr         address in which the UE is listening to trigger procedures
  --ue_port         port in which the UE is listening to trigger procedures
  --scheme          [https|http]
  --n3iwf_address   address of the n3iwf
  --supi_or_suci    unique subscriber identifier
  --k               key
  --opc             operator code
  --ike_bind_addr   address to bind UE IKE to

I did not run the non-3GPP access on a different hardware than the 5G core network, but i believe that once that Y1 and Y2 interfaces are correctly established, you need only create the ipsec link required between  UE and the N3IWF, and then execute the UE and trigger initial registration procedure.

I hope you can solve your problem.

Regards,
Mario

**Reply by dayal007** [reply]
Hi @mariotlemes,

Basically, I want to run the UE in a different machine and the core network in a different machine. So that I can have multiple ue's connected to the core network through n3iwf.

And also I want the UE to be capable of connecting to RAN and also n3iwf. Because why do you think a UE just need to talk to wi-fi all the time? Because n3iwf should support mobility between indoor and outdoor.

And also I dont understand the IoT use case "https://github.com/LABORA-INF-UFG/SBrT2020-Minicurso1/blob/master/docs/non3gpp-iot-use-case/non3gpp-iot-use-case.md". Because what is the purpose of a LoRa Gateway talking to n3iwf? There is no use case right? 

Thanks,
Udhay

**Reply by mariotlemes** [reply]
Hello @dayal007 ,

> Basically, I want to run the UE in a different machine and the core network in a different machine. So that I can have multiple ue's connected to the core network through n3iwf.

Yes, it is possible. You need to verify the ip address of N3IWF for configure the IPSec tunnel between UE and the N3IWF. Again, once that Y1 and Y2 interfaces are correctly established, you need only create the ipsec link required between UE and the N3IWF, and then execute the UE and trigger initial registration procedure.

> And also I want the UE to be capable of connecting to RAN and also n3iwf. Because why do you think a UE just need to talk to wi-fi all the time? Because n3iwf should support mobility between indoor and outdoor.

I'm sorry, but  i'm not think that "an UE just need to talk to wi-fi all the time". I did not write this information. Otherwise, my apologies by the confusion. 

An UE may be connected via NG-RAN or N3IWF (untrusted access), or via both access, simultaneously. The N3IWF is agnostic access, i.e., besides the Wi-Fi network, others non-3GPP access network are possible (Bluetooh, LoRa, and other non-3GPP access network).

> And also I dont understand the IoT use case "https://github.com/LABORA-INF-UFG/SBrT2020-Minicurso1/blob/master/docs/non3gpp-iot-use-case/non3gpp-iot-use-case.md". Because what is the purpose of a LoRa Gateway talking to n3iwf? There is no use case right?

In case of the LoRa network, the end devices, e.g., LoRa sensors, does not have IP stack. For connection with the N3IWF an traditional IP stack is required. Therefore, the loRa gateway is responsible to establish the integration between the non-3GPP access network and the N3IWF.  To help, in this [reference,](https://arxiv.org/pdf/2006.10409.pdf) you find an study that provide the details about the integration of a 5G networks and LoRa access networks.

Regards,
Mario

**Reply by dayal007** [reply]
Hi @mariotlemes,

Thanks for the clarification.

> Yes, it is possible. You need to verify the ip address of N3IWF for configure the IPSec tunnel between UE and the N3IWF. Again, once that Y1 and Y2 interfaces are correctly established, you need only create the ipsec link required between UE and the N3IWF, and then execute the UE and trigger initial registration procedure.

Does the configuration changes I proposed in my first post are sufficient? Or do I need to make additional changes to get this working? I can figure out myself but since you know better. And I am trying to save time for me and others who may need the same thing.

> An UE may be connected via NG-RAN or N3IWF (untrusted access), or via both access, simultaneously. The N3IWF is agnostic access, i.e., besides the Wi-Fi network, others non-3GPP access network are possible (Bluetooh, LoRa, and other non-3GPP access network).

Yes, does the UE you have support both are just one? If it supports both then we can use it for many use cases.

> In case of the LoRa network, the end devices, e.g., LoRa sensors, does not have IP stack. For connection with the N3IWF an traditional IP stack is required. Therefore, the loRa gateway is responsible to establish the integration between the non-3GPP access network and the N3IWF. To help, in this reference, you find an study that provide the details about the integration of a 5G networks and LoRa access networks.

Yes, I understand that. What is the purpose/use-case of the LoRa network to connect to 5g core through n3iwf? The LoRa gateway will be in a stable location and what mobility advantages or any other advantages does it have to connect to 5g core?

Thanks,
Udhay

**Reply by mariotlemes** [reply]
Hello,

I do not know if it is enough because i did not run the test that you mentioned.

I hope that you can solve your problem.

Regards,
Mario

**Comment by manoj1919** [comment]
Hi @dayal007,
 did you figure it out? I am trying to do the same thing, running UE on a different machine. I am not even sure how to connect to n3iwf.
steps you followed will be of great help.
Thanks,
Manoj
